To: vim_dev@googlegroups.com
Subject: Patch 7.3.289
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.289
Problem:    Complete function isn't called when the leader changed.
Solution:   Call ins_compl_restart() when the leader changed.  (Taro Muraoka)
Files:	    src/edit.c


*** ../vim-7.3.288/src/edit.c	2011-05-19 17:25:36.000000000 +0200
--- src/edit.c	2011-08-28 15:53:47.000000000 +0200
***************
*** 3367,3372 ****
--- 3367,3381 ----
      ins_bytes(compl_leader + ins_compl_len());
      compl_used_match = FALSE;
  
+ #ifdef FEAT_COMPL_FUNC
+     /*
+      * To call eval 'completefunc' when leader is changed, restart completion
+      * every time.
+      */
+     if (ctrl_x_mode == CTRL_X_FUNCTION || ctrl_x_mode == CTRL_X_OMNI)
+         ins_compl_restart();
+ #endif
+ 
      if (compl_started)
  	ins_compl_set_original_text(compl_leader);
      else
*** ../vim-7.3.288/src/version.c	2011-08-28 16:00:14.000000000 +0200
--- src/version.c	2011-08-28 16:01:22.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     289,
  /**/

-- 
THEOREM: VI is perfect.
PROOF: VI in roman numerals is 6.  The natural numbers < 6 which divide 6 are
1, 2, and 3. 1+2+3 = 6.  So 6 is a perfect number.  Therefore, VI is perfect.
QED
						    -- Arthur Tateishi

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
